packages/rest/src/coercion/coerce-parameter.ts

```diff
@@ -169,10 +169,37 @@ function coerceObject(input: string | object, spec: ParameterObject) {
   if (typeof data !== 'object' || Array.isArray(data))
     throw RestHttpErrors.invalidData(input, spec.name);
 
-  // TODO(bajtos) apply coercion based on properties defined by spec.schema
+  const schema = extractSchemaFromSpec(spec);
+  if (schema) {
+    // apply coercion based on properties defined by spec.schema
+    const ajv = new Ajv({coerceTypes: true});
+    const validate = ajv.compile(schema);
```

Comment:
We should not create the `Ajv` instance per call. It's fairly expensive.

Reply:
True, I refactored to use one global instance. Or should I use a singleton scope provider instead?